fs/sync/sync_test.go

```diff
@@ -947,6 +947,28 @@ func testServerSideMove(t *testing.T, r *fstest.Run, withFilter, testDeleteEmpty
 	}
 }
 
+// Test move
+func TestMove(t *testing.T) {
+	r := fstest.NewRun(t)
+	defer r.Finalise()
+	file1 := r.WriteFile("sub dir/hello world", "hello world", t1)
+	file2 := r.WriteFile("nested/sub dir/file", "nested", t1)
+	r.Mkdir(r.Fremote)
+
+	// run move with --delete-empty-src-dirs
+	err := MoveDir(r.Fremote, r.Flocal, true)
```

Comment:
I'd quite like to see another test without `--delete-empty-src-dirs`, something like this which I wrote convincing myself that your patch worked :-)

```diff
diff --git a/fs/sync/sync_test.go b/fs/sync/sync_test.go
index 57b62d5a..d05acd9d 100644
--- a/fs/sync/sync_test.go
+++ b/fs/sync/sync_test.go
@@ -948,7 +948,7 @@ func testServerSideMove(t *testing.T, r *fstest.Run, withFilter, testDeleteEmpty
 }
 
 // Test move
-func TestMove(t *testing.T) {
+func TestMoveWithDeleteEmptySrcDirs(t *testing.T) {
 	r := fstest.NewRun(t)
 	defer r.Finalise()
 	file1 := r.WriteFile("sub dir/hello world", "hello world", t1)
@@ -969,6 +969,32 @@ func TestMove(t *testing.T) {
 	fstest.CheckItems(t, r.Fremote, file1, file2)
 }
 
+// Test move
+func TestMoveWithoutDeleteEmptySrcDirs(t *testing.T) {
+	r := fstest.NewRun(t)
+	defer r.Finalise()
+	file1 := r.WriteFile("sub dir/hello world", "hello world", t1)
+	file2 := r.WriteFile("nested/sub dir/file", "nested", t1)
+	r.Mkdir(r.Fremote)
+
+	// run move without --delete-empty-src-dirs
+	err := MoveDir(r.Fremote, r.Flocal, false)
+	require.NoError(t, err)
+
+	fstest.CheckListingWithPrecision(
+		t,
+		r.Flocal,
+		nil,
+		[]string{
+			"sub dir",
+			"nested",
+			"nested/sub dir",
+		},
+		fs.GetModifyWindow(r.Flocal),
+	)
+	fstest.CheckItems(t, r.Fremote, file1, file2)
+}
+
 // Test a server side move if possible, or the backup path if not
 func TestServerSideMove(t *testing.T) {
 	r := fstest.NewRun(t)
```